Feat #22076: Add support for new icon components that are already in the design system

perryfardella opened the pull request:
### What does it do?

Addresses [this](https://github.com/strapi/strapi/issues/22076) issue

Utilises new icon components int he file system to show correct icons for xls, csv and zip files. We do this already for PDF's so everywhere we show the PDF icon specifically, I've added support to show xls, csv and zip file icons also.

### Why is it needed?

Currently we just show a generic file icon for xls, csv and zip files, when we have specific icons available to us in the design system.

### How to test it?

This can be tested by uploading xls, csv and zip files and observing that they display specific icons for those file types as opposed to the generic file icons they show how.

Upload before:

<img width="836" alt="image" src="https://github.com/user-attachments/assets/46115bc9-d792-4613-9919-cb0dcecd5bfc" />


Upload after: 

<img width="834" alt="image" src="https://github.com/user-attachments/assets/9772c173-495a-4d3f-a32f-72a21a0478e8" />


Media library before:

<img width="1590" alt="image" src="https://github.com/user-attachments/assets/7e73392f-3c93-49a1-bbcb-b99fea1e9483" />


Media library after:

<img width="1577" alt="image" src="https://github.com/user-attachments/assets/d0ed280b-bbd8-4cfa-b0d1-77ea78273e30" />


### Related issue(s)/PR(s)

Not related to any other issues/pr(s) I know of.

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/strapi?pullRequest=22617) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/strapi?pullRequest=22617) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/strapi?pullRequest=22617) it.</sub>

perryfardella commented:
I think the SonarCloud Code Analysis can be ignored, it's complaining that this is duplicated code... 
<img width="530" alt="image" src="https://github.com/user-attachments/assets/0c4706fe-8717-4e62-b3d0-6546f6ce3daf" />

lucasboilly commented:
Hi @perryfardella, thanks a lot! 

Unfortunately I'm getting an error when I try to start an app from your branch. I'm gonna share the PR to the team anyway :)

lucasboilly commented:
@perryfardella it seems the errors happens only if I'm using `yarn setup` before using `yarn develop`. This time I started with `yarn develop` and it worked fine. I could check the design and that's all good 👌

I'm joining the errors just in case:
- After `yarn setup`
https://github.com/user-attachments/assets/1e75d475-b991-403e-a9a9-ec1f25f6595c
- After `yarn setup` and `yarn develop`
![CleanShot 2025-01-13 at 13 02 56@2x](https://github.com/user-attachments/assets/e451dc12-ed57-423f-96e1-7e2b131f5ee9)

lucasboilly commented:
@perryfardella all good! I don't have the error anymore :)

I noticed something else while testing. When we click on a file, the preview inside of the modal still shows the default icon. Is it possible to make the right icon appear?

https://github.com/user-attachments/assets/00e3e6d5-7462-459d-8c89-147f628c97c3

perryfardella commented:
@lucasboilly Yep I should be able to correct that, leave it with me. Will move this pr back into draft until I solve that issue.

lucasboilly commented:
@perryfardella perfect!

perryfardella commented:
@lucasboilly  I can see what's happening, we haven't implemented this for PDF's yet either, which is why I missed it. 

<img width="835" alt="image" src="https://github.com/user-attachments/assets/e56961c0-8a38-4a9a-8609-9684867096aa" />

perryfardella commented:
@lucasboilly This is now fixed and I've returned the pr to ready for review.
I've added evidence to the pr description of the asset preview icons working for the new file types.

lucasboilly commented:
@perryfardella it's all good! Thanks a lot :)

simotae14 commented on packages/core/upload/admin/src/components/AssetCard/DocAssetCard.tsx:
```diff
@@ -1,7 +1,7 @@
-import { File, FilePdf } from '@strapi/icons';
 import { useIntl } from 'react-intl';
 import { styled } from 'styled-components';
+import { getFileIconComponent } from '../../utils/icons';
```
you need to add an empty line between the styled import and the getFileIconComponent import

perryfardella replied:
Done.

simotae14 commented on packages/core/upload/admin/src/components/EditAssetDialog/PreviewBox/AssetPreview.tsx:
```diff
@@ -3,11 +3,13 @@ import * as React from 'react';
 
 import { AssetType } from '../../../constants';
+import { getFileIconComponent } from '../../../utils/icons';
+import { typeFromMime } from '../../../utils';
```
this import should occur before import of `../../../utils/icons`

perryfardella replied:
Will action this, Is this a practical consideration or just for styling?

perryfardella replied:
Done.

simotae14 commented on packages/core/upload/admin/src/components/EditAssetDialog/PreviewBox/AssetPreview.tsx:
```diff
@@ -3,11 +3,13 @@ import * as React from 'react';
 import MuxPlayer from '@mux/mux-player-react';
 import { Box, Flex, Typography } from '@strapi/design-system';
-import { File, FilePdf } from '@strapi/icons';
+import { File } from '@strapi/icons';
```
Now this import is useless because you are no more using File

perryfardella replied:
Removed.

simotae14 commented on packages/core/upload/admin/src/components/MediaLibraryInput/Carousel/CarouselAsset.tsx:
```diff
@@ -8,6 +7,7 @@ import { AudioPreview } from '../../AssetCard/AudioPreview';
 import { VideoPreview } from '../../AssetCard/VideoPreview';
 
 import type { File as FileAsset } from '../../../../../shared/contracts/files';
+import { getFileIconComponent } from '../../../utils/icons';
```
There should be at least one empty line between import groups

simotae14 replied:
And this import should occur before import of `../../AssetCard/AudioPreview`

perryfardella replied:
I think my understanding of imports is a bit too rudimentary as I didn't realise this was a problem. Can you point me in the right direction of some resources I should brush up on to make sure I handle imports correctly in the future?

perryfardella replied:
Actually, ignore this. I just read the linting rules and worked out what we're doing here. Everything has been fixed.

simotae14 commented on packages/core/upload/admin/src/utils/tests/typeFromMime.test.ts:
```diff
@@ -14,6 +14,30 @@ describe('typeFromMime', () => {
+    expect(type).toBe(DocType.Zip);
+  });
+
   it('gives a type of document when the mime is neither video nor image', () => {
```
this test is failing right now
`typeFromMime › gives a type of document when the mime is neither video nor image

    expect(received).toBe(expected) // Object.is equality

    Expected: "doc"
    Received: "pdf"`

simotae14 commented on packages/core/upload/admin/src/utils/icons.ts:
```diff
@@ -0,0 +1,18 @@
+import { File, FileCsv, FilePdf, FileXls, FileZip } from '@strapi/icons';
+import { DocType } from '../constants';
+
+export const FILE_TYPE_ICON_COMPONENT_MAP: Record<string, React.ComponentType<any>> = {
```
Please find a way to avoid using any

simotae14 commented on packages/core/upload/admin/src/utils/icons.ts:
```diff
@@ -0,0 +1,18 @@
+  [DocType.Zip]: FileZip,
+} as const;
+
+export const DEFAULT_FILE_ICON: React.ComponentType<any> = File;
```
same here

perryfardella commented:
Thanks for the review @simotae14 
All comments seem very reasonable, will try get these fixed within the next week.

perryfardella commented on packages/core/upload/admin/src/utils/typeFromMime.ts:
Agreed, this is a much simpler way to handle the function logic, I've implemented this change. Cheers!

perryfardella commented on packages/core/upload/admin/src/components/AssetCard/AssetCard.tsx:
```diff
@@ -1,4 +1,4 @@
-import { AssetType } from '../../constants';
```
Had a circular dependency issue, have moved the enums from contants.ts to their own file enums.ts

perryfardella commented on packages/core/upload/admin/src/utils/icons.ts:
```diff
@@ -0,0 +1,33 @@
+import { DocType } from '../enums';
+
+// The IconProps interface is not exported from our design library package, so we need to redefine it here
+interface IconProps extends Omit<SVGProps<SVGSVGElement>, 'fill' | 'stroke'> {
```
Needed to redefine IconProps here to avoid using any typing.

HichamELBSI replied:
You are right, we should definitely export the type.

perryfardella commented:
@simotae14 @lucasboilly 

Is this likely to get re-reviewed and merged or should I close this pr?

HichamELBSI reviewed (changes requested):
Hello @perryfardella, thanks for the contribution, we are finally reviewing the PR. Let us know if you need help on it. Feel free to ping me if you have any question. Let's move forward with your PR so we can merge it 👍🏽

HichamELBSI commented:
Can you squash your commits please

HichamELBSI commented:
@perryfardella Sorry for that, can you please solve the conflicts. I'll watch the PR so we can merge it asap.

HichamELBSI commented:
@perryfardella Your commit message does not follow the standard commit format. Please make sure to run all necessary checks for lint, git, tests, and e2e tests before pushing, to avoid spending too much time on this PR. https://github.com/strapi/strapi/blob/develop/CONTRIBUTING.md

perryfardella commented:
@HichamELBSI Sorry, I've cleaned up the commit name, and fixed a build and typing error, all tests/build/lint now pass.

I'm seeing one e2e test fail, but I confirmed that is also failing on main, so unrelated to my pr (may even just be a local flake).
